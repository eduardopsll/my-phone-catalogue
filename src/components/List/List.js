import React from "react";
import { useSelector, useDispatch } from "react-redux";

import { phoneListSelector } from "../../redux/selectors";
import { ListItem } from "../ListItem";

import styles from "./List.module.scss";
import { setPhoneSelected } from "../../redux/actions";

function List() {
  const list = useSelector(phoneListSelector);
  const dispatch = useDispatch();

  return (
    <ul className={styles.list}>
      {list.map(item => (
        <ListItem key={item.id} data={item} selectItemFn={id => dispatch(setPhoneSelected(id))} />
      ))}
    </ul>
  );
}

export default List;
