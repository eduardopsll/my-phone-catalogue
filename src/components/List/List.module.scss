@import "../../styles/base";

.list {
  flex-grow: 1;
  height: 100%;
  list-style: none;
  margin: 0;
  padding: 0 $space-4;
}
