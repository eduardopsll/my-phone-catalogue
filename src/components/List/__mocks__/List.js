import React from "react";

function List() {
  return <div data-testid="list"></div>;
}

export default List;
