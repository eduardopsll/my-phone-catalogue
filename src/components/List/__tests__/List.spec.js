import React from "react";
import ReactDOM from "react-dom";
import { useSelector } from "react-redux";

import List from "../List";
import phonesMock from "../../../mocks/phones.json";
jest.mock("react-redux");
useSelector.mockImplementation(jest.fn());

describe("List", () => {
  describe("List with items", () => {
    beforeEach(() => {
      useSelector.mockImplementation(selectorFn => selectorFn({ phoneList: phonesMock }));
    });

    it("should render without crashing", () => {
      const div = document.createElement("div");
      ReactDOM.render(<List />, div);
    });
  });

  describe("List without items", () => {
    beforeEach(() => {
      useSelector.mockImplementation(selectorFn => selectorFn({ phoneList: [] }));
    });

    it("should render without crashing", () => {
      const div = document.createElement("div");
      ReactDOM.render(<List />, div);
    });
  });
});
