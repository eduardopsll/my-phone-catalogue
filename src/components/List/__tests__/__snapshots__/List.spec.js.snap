// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List with items should render without crashing 1`] = `
<div>
  <ul
    class="list list--selected"
  >
    <li>
      <a
        class="list-item "
        href="/0"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            iPhone 7
            <span
              class="list-item__text-subtitle"
            >
              Apple
            </span>
          </div>
          <div
            class="list-item__price"
          >
            769
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/1"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            Galaxy S7
            <span
              class="list-item__text-subtitle"
            >
              Samsung
            </span>
          </div>
          <div
            class="list-item__price"
          >
            209
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/2"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            Honor 10
            <span
              class="list-item__text-subtitle"
            >
              Huawei
            </span>
          </div>
          <div
            class="list-item__price"
          >
            399
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/3"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            P10 Lite
            <span
              class="list-item__text-subtitle"
            >
              Huawei
            </span>
          </div>
          <div
            class="list-item__price"
          >
            249
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/4"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            Nokia 7.1
            <span
              class="list-item__text-subtitle"
            >
              Nokia
            </span>
          </div>
          <div
            class="list-item__price"
          >
            275
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/5"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            Zen Phone 5
            <span
              class="list-item__text-subtitle"
            >
              Asus
            </span>
          </div>
          <div
            class="list-item__price"
          >
            359
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/6"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            MI A2
            <span
              class="list-item__text-subtitle"
            >
              Xiaomi
            </span>
          </div>
          <div
            class="list-item__price"
          >
            179
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
    <li>
      <a
        class="list-item "
        href="/7"
      >
        <div
          class="list-item__info-wrapper"
        >
          <img
            alt="phone-icon"
            class="list-item__phone-icon"
            src="phone.svg"
          />
          <div
            class="list-item__text-wrapper"
          >
            Moto G6
            <span
              class="list-item__text-subtitle"
            >
              Motorola
            </span>
          </div>
          <div
            class="list-item__price"
          >
            199
            €
          </div>
        </div>
        <img
          alt="chevron-icon"
          class="list-item__chevron-icon"
          src="chevron.svg"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`List List without items should render without crashing 1`] = `
<div>
  <ul
    class="list list--selected"
  />
</div>
`;
